README.md

```diff
@@ -1,55 +1,185 @@
+be provided to our inputs wide open.
+
+With React, we can do much better! Consider this `TextField` component:
 
 ```javascript
 const defaultLimit = 100
 const noop = () => {}
 
 export default class TextField extends React.Component {
   render() {
-    // assume that we have some of these classes defined
-    return <input className="field field-light"
+    // Assume that we have the css classes defined elsewhere.
+    return <input
+             className="field field-light"
              onChange={this.props.onChange || noop}
-             limit={this.props.limit || defaultLimit} />
+             maxLength={this.props.limit || defaultLimit} />
   }
 }
 ```
 
-Now we can do whatever we want with `<TextField />`.
+First off, notice that here again, what we have a component that
+fits the presentational pattern. It's a simple wrapper around an `<input>`. But
+look how powerful it is! This simple wrapper establishes the CSS classes we will
+use in one place for every single input used throughout the app. Think of how
+easy it would be to change if we later decided we wanted a different look! But
+that's not all we've accomplished here. The component also establishes a
+straightforward API for all our text fields consisting of an `onChange` callback
+-- because in most cases our `<input>`s are going to need to perform some action
+when the users type -- and a `limit` for the amount of characters that a user
+can enter in the field. So although our presentational component is simple, it
+can still have a degree of interactivity through the addition of callbacks.
 
-Why not just have a `<input className="field field-light" {...props} />`
-and call it a day? I'm sure some would disagree, but I think limiting the
-`TextField` props API is a win for simplicity with the application. In the
-example above, we can change the `limit` and the `onChange` handler — nothing
-else. If we need to change something else, maybe we need another component or
-another abstraction.
+Now, of course, we can argue about whether wrapping the `<input>` field in this
+way is a good idea. After all, `<input>`s are nice simple implementations in their
+own right. But providing a component-based interface to text inputs as we have in
+the field above is potentially a great win for simplicity in our app. It specifically
+defines what we mean by a text input; it does so in a way that arguably covers
+the majority of use-cases we can imagine for a simple text input; and it provides
+this definition in one place that can be found and updated easily in the future.
+Win, win, win. Are we beginning to see the power off presentational components? 
+Good.
 
-It's easy to go from here to building a `<Header />`, `<Footer />` etc.
+Now imagine that it's not just the `TextField` that our team has executed
+in this way, but say we've also defined a `<Header />` and a `<Footer />`, as
+well, of course, as other more unique and customized modules that are still
```

Comment:
I feel like removing `, of course,` improves the readability a lot, the 'of course' breaks up the 'as well as' a little too much (might just be me)